pynsee/macrodata/_get_metadata.py

```diff
@@ -94,10 +94,10 @@ def _get_dataset_metadata_core(dataset, update=False, silent=False):
         [idbank_list_dataset, df_cleflow_splitted], axis=1
     )
 
-    n_dimensions = len(dataset_dimension.index)
+    for irow, dim_id in dataset_dimension["dimension"].items():
+        if dim_id not in idbank_list_dataset.columns:
+            continue

```

Comment:
was there an issue that required this? if so, what was it and on which dataset?

Reply:
I'm not sure I got everything... do you mean that you don't know whether this is required or not?

Reply:
No, I couldn't isolate a specific case (that being said, I didn't check thorougly). But as #256 's title is _metadata download in macrodata module should not retrieve metadata on all datasets but only on one_, I figure this happened indeed. Maybe that was in the past.

What's for sure right now, is that the check already exists, but at the end of the loop execution here: https://github.com/InseeFrLab/pynsee/blob/31f0096b687e3f2282136911a143d2404cdf373a/pynsee/macrodata/_get_metadata.py#L118. So even if this is a legacy security, we should bring it earlier.